app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarScreen.kt

```diff
@@ -291,6 +292,80 @@ enum class TaskFilter(val displayName: String) {
     Meal("Meal")
 }
 
+@Composable
+fun StepCounter(
+    model: CalendarViewModel = hiltViewModel(),
+    date: LocalDate
+) {
+    // Required for checking whether user has permissions before entering the window,
+    // as users can revoke permissions at any time
+    val permissionContract = PermissionController.createRequestPermissionResultContract()
+
+    val launcher =
+        rememberLauncherForActivityResult(permissionContract) {
+            model.getDaysSteps(date)
+        }
+
+    LaunchedEffect(Unit) {
+        if (model.permissionsGranted()) {
+            model.getDaysSteps(date)
+        } else {
+            launcher.launch(PERMISSIONS)
+        }
+    }
+
+    val steps = model.stepsCount.collectAsStateWithLifecycle()
+    val goal = 10000f // TODO create way for user to set a step goal?
+    val progress = steps.value / goal
+
+
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = Modifier.fillMaxSize()
+    ) {
+        // These have to be set here because for some reason MaterialTheme.colorScheme
+        // can't be used inside a Canvas ¯\_(ツ)_/¯
```

Comment:
Comment seems redundant.

If I recall correctly, `colorScheme` is a composable. Compiler should show that exact problem - that a composable must be called within other composable function, which Canvas is not, if I am not mistaken.

Either way, an error should be shown by the compiler, if developer would not understand why this is moved to outer space and would move it in. So, I believe, this is unnecessary.

Reply:
Yes, it throws that error, but

![image](https://github.com/PVP-K410/Calendar/assets/25864361/0d4a29fb-afd2-4839-9e77-bfd0131444b7)

Doesn't the @Composable mean it's a composable?

Reply:
Yes, but you are providing code for lambda context, which is not a @Composable, but a simple DrawScope instance method.

If it was `DrawScope.@Composable() -> Unit`, then it should work.

Reply:
Ah, makes sense. Thanks